app/src/main/java/com/example/androidtrainingproject/login/LoginScreen.kt

```diff
@@ -3,113 +3,154 @@ package com.example.androidtrainingproject.login
 import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.width
-import androidx.compose.material3.Button
 import androidx.compose.material3.IconButton
 import androidx.compose.material3.OutlinedTextField
 import androidx.compose.material3.Text
+import androidx.compose.material3.TextFieldDefaults
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.paint
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.res.dimensionResource
 import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.text.input.PasswordVisualTransformation
 import androidx.compose.ui.text.input.VisualTransformation
-import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import androidx.hilt.navigation.compose.hiltViewModel
 import com.example.androidtrainingproject.R
+import com.example.androidtrainingproject.ui.shared.WideButton
+import com.example.androidtrainingproject.ui.theme.ErrorDarkRed
+import com.example.androidtrainingproject.ui.theme.Purple
 
 
 @OptIn(ExperimentalComposeUiApi::class)
 @Composable
 fun LoginScreen(navigateToProductDetails: () -> Unit = {}) {
-    val usernameField = remember { mutableStateOf("") }
-    val passwordField = remember { mutableStateOf("") }
+    var usernameField by remember { mutableStateOf("") }
+    var passwordField by remember { mutableStateOf("") }
 
-    val isPasswordVisible = remember { mutableStateOf(false) }
+    var markInputsAsWrong by remember { mutableStateOf(false) }
+    var isPasswordVisible by remember { mutableStateOf(false) }
 
     val loginViewModel: LoginViewModel = hiltViewModel()
+    val loginFailure by loginViewModel.loginFailure.collectAsState()
 
     val passwordIcon = isPasswordVisible.let {
-        if (isPasswordVisible.value) {
+        if (isPasswordVisible) {
             R.drawable.ic_visibility_on
         } else {
             R.drawable.ic_visibility_off
         }
     }
 
+    val context = LocalContext.current
+    val loginFailureMessage = stringResource(id = R.string.login_failure)
     val keyboardController = LocalSoftwareKeyboardController.current
 
-    Column(modifier = Modifier.padding(10.dp)) {
+    LaunchedEffect(loginFailure) {
```

Comment:
Note: You can use a `SharedFlow` for that so it doesn't emit an unnecessary boolean that you have to evaluate.